x/ibc/light-clients/07-tendermint/types/upgrade.go

```diff
@@ -20,14 +23,24 @@ import (
 //   and ProofSpecs do not match parameters set by committed client
 func (cs ClientState) VerifyUpgrade(
 	ctx sdk.Context, cdc codec.BinaryMarshaler, clientStore sdk.KVStore,
-	upgradedClient exported.ClientState, proofUpgrade []byte,
+	upgradedClient exported.ClientState, upgradeHeight exported.Height, proofUpgrade []byte,
 ) error {
-	if cs.UpgradePath == nil {
+	if cs.UpgradePath == "" {
 		return sdkerrors.Wrap(clienttypes.ErrInvalidUpgradeClient, "cannot upgrade client, no upgrade path set")
 	}
+	// construct MerklePath from upgradePath
+	upgradeKeys := strings.Split(cs.UpgradePath, "/")
+	// append upgradeHeight to last key in merkle path
+	upgradeKeys[len(upgradeKeys)-1] = fmt.Sprintf("%s/%d", upgradeKeys[len(upgradeKeys)-1], upgradeHeight.GetEpochHeight())
```

Comment:
why dont use just actually `append(upgradeKeys, fmt.Sprintf("%d. upgradeHeight.GetEpochHeight())` I think it'd improve readability.

The above creates a strange slice imo. All keys except for the last will not contain `/` since they are split. but in the last you insert in the last key as `lastkey/height`. This seems odd since it relies on `NewMerklePath` operating a certain way (ie allowing the possibility of  `/` in each key)

Reply:
`append(upgradeKeys, fmt.Sprintf("%d. upgradeHeight.GetEpochHeight())` wouldn't be correct because the height is not a separate MerkleKey. `upgradedClient/{height}` is the last MerkleKey which is what the code shows.

> The above creates a strange slice imo. All keys except for the last will not contain / since they are split. but in the last you insert in the last key as lastkey/height. 

This is actually how every SDK key gets converted to MerklePath in practice. A typical SDK key like: `ibc/clients/{clientID}/consensusState/10` is actually converted to two keys:

`["ibc", "clients/{clientID}/consensusState/10"]`

This is just recreating the last key manually.

Reply:
Yes good question, any `/` within a key must be URL-escaped. I added the correct unescaping logic so it all works. This is documented in `CreateClient` client tx, but will have to add a lot of documentation on upgrade functionality after code-freeze.

I couldn't add tests for this directly without making the upgradedClient key contain a `/` which I don't want to do for simplicity. However, I tested this manually on my own local code and everything works as expected